docs/ingress-gateway-asm-manifests/base/deployment-service.yaml

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START servicemesh_base_deployment_service_serviceaccount_asm_ingressgateway]
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: asm-ingressgateway
+# [END servicemesh_base_deployment_service_serviceaccount_asm_ingressgateway]
+---
+# [START servicemesh_base_deployment_service_deployment_asm_ingressgateway]
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: asm-ingressgateway
+spec:
+  selector:
+    matchLabels:
+      asm: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        asm: ingressgateway
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+        env:
+        - name: ISTIO_META_UNPRIVILEGED_POD
+          value: "true"
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+        resources:
+          limits:
+            cpu: 2000m
+            memory: 1024Mi
+          requests:
+            cpu: 100m
+            memory: 128Mi
+        securityContext:
```

Comment:
thinking back to the discussion in asm packages https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/pull/1244

Reply:
Yeah, good discussion in there, but it was more about migrating existing customer leveraging this manifests (changes on the `Service` will cause downtime). In the case of this PR here, I'm not changing that, both `Deployment` and `Service` remain the same, I just moved them under the same file now.
This PR also adds HPA, Role and RoleBinding.
Does that clarify? Let me know if you have any other feedback/question. Thanks!

Reply:
makes sense, thanks 👍